Manual/Releases/v4_22_0.lean

```diff
@@ -21,6 +21,202 @@ file := "v4.22.0"
+  Handles repeated simp calls correctly (e.g., inside `all_goals`), but skips macros.
+
+* **Detection of possibly looping lemmas**
+
+  [#8865](https://github.com/leanprover/lean4/pull/8865) allows `simp` to recognize and warn about simp lemmas that are
+  likely looping in the current simp set. It does so automatically
+  whenever simplification fails with the dreaded “max recursion depth”
+  error, but it can be made to do it always with `set_option
+  linter.loopingSimpArgs true`. This check is not on by default because it
+  is somewhat costly, and can warn about simp calls that still happen to
+  work.
+
+* **Faster `simp` via cache reuse**
+
+  [#8880](https://github.com/leanprover/lean4/pull/8880) makes `simp` consult
+  its own cache more often, to avoid replicating work.
+
+* **Explicit `defeq` attribute for `dsimp`**
+
+  [#8419](https://github.com/leanprover/lean4/pull/8419) introduces an explicit `defeq` attribute to mark theorems that
+  can be used by `dsimp`. The benefit of an explicit attribute over the
+  prior logic of looking at the proof body is that we can reliably omit
+  theorem bodies across module boundaries. It also helps with intra-file
+  parallelism.
+
+### Named errors with explanations
+Lean now supports named error messages with associated explanations.
+
+[#8649](https://github.com/leanprover/lean4/pull/8649) and [#8730](https://github.com/leanprover/lean4/pull/8730) add
+macro syntax for registering
+and throwing named errors, mechanisms for displaying error names in the Infoview and at the command line,
+and the ability to link to error explanations in the manual (once the explanations are added).
+
+This infrastructure lays the foundation for a searchable error index and improved diagnostics.
+
+### `finally` section
+[#8723](https://github.com/leanprover/lean4/pull/8723) implements a `finally` section following a (potentially empty)
+`where` block. `where ... finally` opens a tactic sequence block in
+which the goals are the unassigned metavariables from the definition
+body and its auxiliary definitions that arise from use of `let rec` and
+`where`.
+This can be useful for discharging multiple proof obligations in the definition body
+by a single invocation of a tactic such as `all_goals`:
+```lean
+example (i j : Nat) (xs : Array Nat) (hi : i < xs.size) (hj: j < xs.size) :=
+  match i with
+  | 0 => x
+  | _ => xs[i]'?_ + xs[j]'?_
+where x := 13
+finally all_goals assumption
+```
+
+### Library highlights
+
+Notable additions to the standard library are:
+
+* Iterators ([#8420](https://github.com/leanprover/lean4/pull/8420), [#8545](https://github.com/leanprover/lean4/pull/8545), [#8615](https://github.com/leanprover/lean4/pull/8615), [#8629](https://github.com/leanprover/lean4/pull/8629), [#8768](https://github.com/leanprover/lean4/pull/8768)),
+
+* Polymorphic ranges ([#8784](https://github.com/leanprover/lean4/pull/8784)) and slices ([#8947](https://github.com/leanprover/lean4/pull/8947)),
```

Comment:
Probably best to show the new syntax here:

```suggestion
* Polymorphic ranges ([#8784](https://github.com/leanprover/lean4/pull/8784)) and slices ([#8947](https://github.com/leanprover/lean4/pull/8947))

  These introduce new syntax for ranges (`1...*`, `1...=3`, `1...<3`, `1<...=2`, `*...=3.`), which can also be used for slices (e.g. `xs[*...end]`).
```

Given that it feels like built-in syntax to the user and the visibility, maybe worth making this its own item, next to “Library highlights”.